core/trino-main/src/main/java/io/trino/operator/aggregation/state/LongDecimalWithOverflowAndLongStateSerializer.java

```diff
@@ -54,20 +66,25 @@ public void deserialize(Block block, int index, LongDecimalWithOverflowAndLongSt
     {
         if (!block.isNull(index)) {
             Slice slice = VARBINARY.getSlice(block, index);
-            if (slice.length() != SERIALIZED_SIZE) {
-                throw new IllegalStateException("Unexpected serialized state size: " + slice.length());
-            }
+            long[] decimal = state.getDecimalArray();
+            int offset = state.getDecimalArrayOffset();
 
-            long count = slice.getLong(0);
-            long overflow = slice.getLong(Long.BYTES);
+            long high = 0;
+            long overflow = 0;
+            long count = 1;
+            if (slice.length() > Long.BYTES) {
```

Comment:
branchless version
```
int sliceLength = slices.length();
long low = slice.getLong(0);
long highOffset = sliceLength == 4 | sliceLength == 2 ? 1 : 0; 
long high = slice.getLong(highOffset) & (highOffset * -1L);
// similar for count & overflow
```
Would be great to learn if JIT actually generates branchless versions